staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go

```diff
@@ -602,9 +654,19 @@ func (e *Store) Update(ctx context.Context, name string, objInfo rest.UpdatedObj
 			return nil, false, err
 		}
 	}
+	success = true // used in the defer'ed cleanup
```

Comment:
same question about success being false in a persisted-yet-AfterCreate-failed scenario

Reply:
we already have a bug because we do deallocate despite the object is persisted, because there a repair controller that will reallocate the resource from the object stored, but during that time another service can take that resource
https://github.com/kubernetes/kubernetes/pull/93199#issuecomment-665157865